scanner/services/indexer.go

```diff
@@ -104,17 +113,28 @@ func (s *indexerService) GetIndexReport(ctx context.Context, req *v4.GetIndexRep
 		"hash_id", req.GetHashId(),
 	)
 	zlog.Info(ctx).Msg("getting index report for container image")
-	ccIR, err := getClairIndexReport(ctx, s.indexer, req.GetHashId())
-	if err != nil {
+	ir, err := s.getIndexReport(ctx, req.GetHashId())
+	switch {
+	case errors.Is(err, errox.NotFound):
 		zlog.Warn(ctx).Err(err).Send()
+	case err != nil:
+		zlog.Error(ctx).Err(err).Msg("internal error")
+	}
+	return ir, err
+}
+
+// getIndexReport fetches the Index Report for the resource with the given hashID.
+// Users of this function may be interested in logging the output.
```

Comment:
nit: it _might_ be helpful to make it clearer about the intentions (i.e., no logging should take place here). Maybe something like:
```suggestion
// getIndexReport fetches the v4.IndexReport for the resource with the given
// hashID. No logging is performed; however, callers of this method may be
// interested in logging the error. Returns errox.NotFound when the report does
// not exist.
```

Reply:
honestly I think I'll take sourcery's suggestion and call this `indexReportNoLog`

Reply:
never mind keeping the name as-is, but will update the comment with your suggestion